engine/src/main/java/io/camunda/zeebe/engine/processing/deployment/model/element/ExecutableCatchEvent.java

```diff
@@ -27,6 +27,8 @@ public interface ExecutableCatchEvent extends ExecutableFlowElement {
 
   boolean isSignal();
 
+  boolean isCompensation();
+
   default boolean isNone() {
     return !isTimer() && !isMessage() && !isError() && !isLink() && !isEscalation() && !isSignal();
```

Comment:
:x: We should exclude the compensation event when checking for none events.

```suggestion
    return !isTimer() && !isMessage() && !isError() && !isLink() && !isEscalation() && !isSignal() && !isCompensation();
```